src/Backend/Modules/Blog/Actions/DeleteCategory.php

```diff
@@ -31,33 +31,32 @@ public function execute(): void
         $this->id = (int) $deleteFormData['id'];
 
         // does the item exist
-        if ($this->id !== 0 && BackendBlogModel::existsCategory($this->id)) {
-            // get data
-            $this->record = (array) BackendBlogModel::getCategory($this->id);
-
-            // allowed to delete the category?
-            if (BackendBlogModel::deleteCategoryAllowed($this->id)) {
-                // call parent, this will probably add some general CSS/JS or other required files
-                parent::execute();
-
-                // delete item
-                BackendBlogModel::deleteCategory($this->id);
-
-                // category was deleted, so redirect
-                $this->redirect(
-                    BackendModel::createURLForAction('Categories') . '&report=deleted-category&var=' .
-                    rawurlencode($this->record['title'])
-                );
-            } else {
-                $this->redirect(
-                    // delete category not allowed
-                    BackendModel::createURLForAction('Categories') . '&error=delete-category-not-allowed&var=' .
-                    rawurlencode($this->record['title'])
-                );
-            }
-        } else {
-            // something went wrong
+        if ($this->id === 0 || !BackendBlogModel::existsCategory($this->id)) {
             $this->redirect(BackendModel::createURLForAction('Categories') . '&error=non-existing');
         }
+
+        // get data
+        $this->record = (array) BackendBlogModel::getCategory($this->id);
+
+        // allowed to delete the category?
+        if (BackendBlogModel::deleteCategoryAllowed($this->id)) {
+            // call parent, this will probably add some general CSS/JS or other required files
+            parent::execute();
+
+            // delete item
+            BackendBlogModel::deleteCategory($this->id);
+
+            // category was deleted, so redirect
+            $this->redirect(
+                BackendModel::createURLForAction('Categories') . '&report=deleted-category&var=' .
+                rawurlencode($this->record['title'])
+            );
+        } else {
```

Comment:
we usually add a `return` below the redirect, the redirect will throw an exception but by adding the return it is more clear that the code stops there (if needed maybe also add that in other places)